autonick.lua

```diff
@@ -14,41 +14,87 @@ If there are fewer available nicknames than dwarves, the remaining
 dwarves will go un-nicknamed.
 
 You may wish to use this script with the "repeat" command, e.g:
-``repeat -name autonick -time 3 -timeUnits months -command [ autonick ]``
+``repeat -name autonick -time 3 -timeUnits months -command [ autonick -f ]``
 
+Usage:
+
+    autonick [<options>]
+
+Options:
+
+:``-f``, ``--force``:
+    skip confirmation. useful for scripting.
+:``-h``, ``--help``:
+    Show this text.
+:``-q``, ``--quiet``:
+    Do not report how many dwarves were given nicknames.
 ]====]
 
-local seen = {}
---check current nicknames
-for _,unit in ipairs(df.global.world.units.active) do
-    if dfhack.units.isCitizen(unit) and
-    unit.name.nickname ~= "" then
-        seen[unit.name.nickname] = true
-    end
-end
+local options = {}
+
+local argparse = require('argparse')
+local commands = argparse.processArgsGetopt({...}, {
+    {'h', 'help', handler=function() options.help = true end},
+    {'f', 'force', handler=function() options.force = true end},
+    {'q', 'quiet', handler=function() options.quiet = true end},
+})
 
-local names = {}
--- grab list, put in array
-local path = dfhack.getDFPath () .. "/dfhack-config/autonick.txt";
-for line in io.lines(path) do
-    line = line:trim()
-    if (line ~= "")
-    and (not line:startswith('#'))
-    and (not seen[line]) then
-        table.insert(names, line)
-        seen[line] = true
+-- autonick does not use commands, so if there are any, the script is being used wrong.
+if #commands ~= 0 then options.help = true end
+
+if options.help == true then
+    print(dfhack.script_help())
+else
```

Comment:
You can avoid indenting the main body of the code by returning in the `if` block:
```
if options.help then
    print(dfhack.script_help())
    return
end

-- no else block needed. the remaining code can be unindented
```